x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/action/TransportXPackUsageAction.java

```diff
@@ -56,12 +56,17 @@ protected void masterOperation(Task task, XPackUsageRequest request, ClusterStat
                 listener.delegateFailure((l, usages) -> l.onResponse(new XPackUsageResponse(usages)));
         final AtomicReferenceArray<Usage> featureSetUsages = new AtomicReferenceArray<>(usageActions.size());
         final AtomicInteger position = new AtomicInteger(0);
-        final BiConsumer<XPackUsageFeatureAction, ActionListener<List<Usage>>> consumer = (featureUsageAction, iteratingListener) ->
-                client.executeLocally(featureUsageAction, request, iteratingListener.delegateFailure((l, usageResponse) -> {
-                    featureSetUsages.set(position.getAndIncrement(), usageResponse.getUsage());
-                    // the value sent back doesn't matter since our predicate keeps iterating
-                    l.onResponse(Collections.emptyList());
-                }));
+        final BiConsumer<XPackUsageFeatureAction, ActionListener<List<Usage>>> consumer = (featureUsageAction, iteratingListener) -> {
+            // Since we're executing the actions locally we should create a new request
+            // to avoid mutating the original request and setting the wrong parent task
+            final XPackUsageRequest childRequest = new XPackUsageRequest();
+            childRequest.setParentTask(request.getParentTask());
```

Comment:
Just to check, does this mean that we exit the iteration early on cancellation (because the parent task would be banned from running any children)? Perhaps add a comment to that effect if so.